common/aws/cli.go

```diff
@@ -3,20 +3,46 @@ package aws
 import (
 	"github.com/Layr-Labs/eigenda/common"
 	"github.com/urfave/cli"
+	"time"
 )
 
 var (
-	RegionFlagName          = "aws.region"
-	AccessKeyIdFlagName     = "aws.access-key-id"
-	SecretAccessKeyFlagName = "aws.secret-access-key"
-	EndpointURLFlagName     = "aws.endpoint-url"
+	RegionFlagName                      = "aws.region"
+	AccessKeyIdFlagName                 = "aws.access-key-id"
+	SecretAccessKeyFlagName             = "aws.secret-access-key"
+	EndpointURLFlagName                 = "aws.endpoint-url"
+	FragmentPrefixCharsFlagName         = "aws.fragment-prefix-chars"
+	FragmentParallelismFactorFlagName   = "aws.fragment-parallelism-factor"
+	FragmentParallelismConstantFlagName = "aws.fragment-parallelism-constant"
+	FragmentReadTimeoutFlagName         = "aws.fragment-read-timeout"
+	FragmentWriteTimeoutFlagName        = "aws.fragment-write-timeout"
 )
 
 type ClientConfig struct {
-	Region          string
-	AccessKey       string
+	// The region to use when interacting with S3. Default is "us-east-2".
+	Region string
+	// The access key to use when interacting with S3.
+	AccessKey string
+	// The secret key to use when interacting with S3.
 	SecretAccessKey string
-	EndpointURL     string
+	// The URL of the S3 endpoint to use. If this is not set then the default AWS S3 endpoint will be used.
+	EndpointURL string
+
+	// The number of characters of the key to use as the prefix for fragmented files.
+	// A value of "3" for the key "ABCDEFG" will result in the prefix "ABC". Default is 3.
+	FragmentPrefixChars int
+	// This framework utilizes a pool of workers to help upload/download files. A non-zero value for this parameter
+	// adds a number of workers equal to the number of cores times this value. Default is 8. In general, the number
+	// of workers here can be a lot larger than the number of cores because the workers will be blocked on I/O most
+	// of the time.
+	FragmentParallelismFactor int
+	// This framework utilizes a pool of workers to help upload/download files. A non-zero value for this parameter
+	// adds a constant number of workers. Default is 0.
+	FragmentParallelismConstant int
```

Comment:
Why not just use a single field to explicitly define the number of workers?

Reply:
This is based on @dmanc's request to scale the number of workers based on the number of cores on the machine. The pattern here allows for the user to specify either a fixed number of threads or a number that varies with the number of cores.

If you think this is over complicated I'm willing to go back to having a constant number of worker threads.